aws/sdk/integration-tests/s3/tests/stalled-stream-protection.rs

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use aws_sdk_s3::config::{Region, StalledStreamProtectionConfig};
+use aws_sdk_s3::primitives::ByteStream;
+use aws_sdk_s3::{Client, Config};
+use bytes::BytesMut;
+use std::error::Error;
+use std::future::Future;
+use std::net::SocketAddr;
+use std::time::Duration;
+use tracing::debug;
+
+// This test doesn't work because we can't count on `hyper` to poll the body,
+// regardless of whether we schedule a wake. To make this functionality work,
+// we'd have to integrate more closely with the orchestrator.
+//
+// I'll leave this test here because we do eventually want to support stalled
+// stream protection for uploads.
```

Comment:
I think it would have to be polled twice:
- once to set the start of polling time used to calculate throughput
- once more to see that throughput is below the minimum

Cases where it gets checked once only won't be marked as failed.